ext/semian/semian_resource_alloc.h

```diff
@@ -0,0 +1,47 @@
+/*
+For memory management operations of semian resources.
+*/
+#ifndef SEMIAN_RESOURCE_ALLOC_H
+#define SEMIAN_RESOURCE_ALLOC_H
+
+#include <semian.h>
```

Comment:
You have a circular reference here. You can probably be more specific in what you need to include.